drivers/wifi/nrf_wifi/src/wpa_supp_if.c

```diff
@@ -931,8 +931,12 @@ int nrf_wifi_wpa_supp_associate(void *if_priv, struct wpa_driver_associate_param
 		assoc_info.use_mfp = NRF_WIFI_MFP_REQUIRED;
 	}
 
-	if (params->bss_max_idle_period) {
-		assoc_info.bss_max_idle_time = params->bss_max_idle_period;
+	if (vif_ctx_zep->bss_max_idle_period_user_set) {
+		assoc_info.bss_max_idle_time = vif_ctx_zep->bss_max_idle_period;
+	} else {
+		if (params->bss_max_idle_period) {
+			assoc_info.bss_max_idle_time = params->bss_max_idle_period;
+		}
```

Comment:
IMHO we should revert this support in the supplicant and move the Kconfig to Wi-Fi management, so, that both build time and runtime are in Wi-Fi management (at least till we find an acceptable solution for supplicant upstream).